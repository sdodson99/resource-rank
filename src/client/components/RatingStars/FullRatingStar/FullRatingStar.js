import React from 'react';
import PropTypes from 'prop-types';
import styles from './FullRatingStar.module.css';
import { StarRounded } from '@material-ui/icons';

const FullRatingStar = ({ size, color }) => (
  <div className={styles.FullRatingStar} data-testid="FullRatingStar">
    <StarRounded htmlColor={color} style={{ fontSize: size }} />
  </div>
);

FullRatingStar.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
};

FullRatingStar.defaultProps = {
  size: 20,
  color: 'black',
};

export default FullRatingStar;
