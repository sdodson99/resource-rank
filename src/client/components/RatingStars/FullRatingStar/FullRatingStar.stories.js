import React from 'react';
import FullRatingStar from './FullRatingStar';

export default {
  component: FullRatingStar,
  title: 'Components/RatingStars/FullRatingStar',
  args: {
    size: 50,
  },
};

const Template = (args) => <FullRatingStar {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Color = Template.bind({});
Color.args = {
  color: 'orange',
};

export const Size = Template.bind({});
Size.args = {
  ...Color.args,
  size: 100,
};
