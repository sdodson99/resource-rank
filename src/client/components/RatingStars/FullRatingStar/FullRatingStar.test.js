import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import FullRatingStar from './FullRatingStar';
import { createRenderer } from 'react-test-renderer/shallow';

describe('<FullRatingStar />', () => {
  it('should mount', () => {
    render(<FullRatingStar />);

    const fullRatingStar = screen.getByTestId('FullRatingStar');

    expect(fullRatingStar).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(<FullRatingStar />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with custom size', () => {
    const component = createRenderer().render(<FullRatingStar size={100} />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with custom color', () => {
    const component = createRenderer().render(
      <FullRatingStar color={'yellow'} />
    );

    expect(component).toMatchSnapshot();
  });
});
