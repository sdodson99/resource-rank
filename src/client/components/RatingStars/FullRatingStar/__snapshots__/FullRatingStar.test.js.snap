// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullRatingStar /> should render correctly 1`] = `
<div
  className="FullRatingStar"
  data-testid="FullRatingStar"
>
  <Memo
    htmlColor="black"
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</div>
`;

exports[`<FullRatingStar /> should render correctly with custom color 1`] = `
<div
  className="FullRatingStar"
  data-testid="FullRatingStar"
>
  <Memo
    htmlColor="yellow"
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</div>
`;

exports[`<FullRatingStar /> should render correctly with custom size 1`] = `
<div
  className="FullRatingStar"
  data-testid="FullRatingStar"
>
  <Memo
    htmlColor="black"
    style={
      Object {
        "fontSize": 100,
      }
    }
  />
</div>
`;
